src/js/Slider.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import ReactSlider from 'rc-slider';
+import 'rc-slider/assets/index.css';
+import store from './store'
+import { updateSliderPosition } from './actions'
+import uiController from './Controllers/UIController'
+
+
+export default class Slider extends React.Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            value: props.sliderData.position,
+            cursor: "auto"
+        }
+        this.getSliderFooter = this.getSliderFooter.bind(this)
+        this.onSliderChange = this.onSliderChange.bind(this)
+        this.onShiftSlider = this.onShiftSlider.bind(this)
+    }
+
+    getSliderFooter() {
+        const sliderData = this.props.sliderData
+        if (sliderData.footer) {
+            if (sliderData.footer.length === 1) { // Static Footer
+                return sliderData.footer[0]
+            }
+            else if (sliderData.footer.length > this.state.value - 1) {
+                return sliderData.footer[this.state.value - 1]
+            }
+        }
+        return null
+    }
+
+    onSliderChange(value) {
+        this.setState({cursor: "grabbing"})
+        if (value < 1 || value > this.props.sliderData.numTicks) { return; }
+        this.setState({ value: value }, () => {
+            uiController.onSliderKeepContext(this.props.sliderData.msgID, this.props.sliderAppId)
```

Comment:
```suggestion
            uiController.onSliderKeepContext(this.props.sliderData.msgID, this.props.sliderAppId, this.props.sliderData.timeout)
```

OnResetTimeout should reset based on the provided timeout. Currently it reverts back to 10 seconds whenever the slider is changed.

(this should work fine if timeout is null)